core/templates/dev/head/pages/exploration_player/AnswerFeedbackPairDirective.js

```diff
@@ -66,6 +66,10 @@ oppia.directive('answerFeedbackPair', [
             }
             return shortAnswerHtml;
           };
+
+          $scope.getRequestForHint = function() {
+            return $scope.data.learnerAnswer;
```

Comment:
Re the "two directives" question, it's a question of what's being conceptually represented by the directive. If you think of the directive as representing a request/response pair (where the request can be an answer or a hint request), then it's fine to use the same directive for both. In fact this is probably preferable in order to avoid duplication.

The thing to ensure is that this doesn't leak all over the place so that there's general confusion between answer/hint/request. It does seem fine to me to have a list containing the sequence of requests/responses but you might want to name that list something different from answerFeedbackPairs if it's going to include hint requests as well. So, you'll need to use some judgement to figure out where in the flow the "answer" part ends and where the "general request/response" part begins, and name things accordingly.